--- build/gyp/pylib/gyp/generator/make.py
+++ build/gyp/pylib/gyp/generator/make.py
@@ -294,7 +294,7 @@
  # to replicate this environment fallback in make as well.
  CC.host ?= %(CC.host)s
  CFLAGS.host ?=
- CXX.host ?= %(CXX.host)s
+ CXX.host ?= %(CXX.target)s
  CXXFLAGS.host ?=
  LINK.host ?= %(LINK.host)s
  LDFLAGS.host ?=
@@ -317,7 +317,7 @@
  # We write to a dep file on the side first and then rename at the end
  # so we can't end up with a broken dep file.
  depfile = $(depsdir)/$(call replace_spaces,$@).d
- DEPFLAGS = -MMD -MF $(depfile).raw
+ # DEPFLAGS = -MMD -MF $(depfile).raw

  # We have to fixup the deps output in a few ways.
  # (1) the file output should mention the proper .o file.


--- build/gyp/pylib/gyp/generator/make.py
+++ build/gyp/pylib/gyp/generator/make.py
@@ -210,19 +210,19 @@
  
  LINK_COMMANDS_AIX = """\
  quiet_cmd_alink = AR($(TOOLSET)) $@
- cmd_alink = rm -f $@ && $(AR.$(TOOLSET)) crs $@ $(filter %.o,$^)
+ cmd_alink = rm -f $@ && $(AR.$(TOOLSET)) -Xany crs $@ $(filter %.o,$^)
  
  quiet_cmd_alink_thin = AR($(TOOLSET)) $@
- cmd_alink_thin = rm -f $@ && $(AR.$(TOOLSET)) crs $@ $(filter %.o,$^)
+ cmd_alink_thin = rm -f $@ && $(AR.$(TOOLSET)) -Xany crsT $@ $(filter %.o,$^)
  
  quiet_cmd_link = LINK($(TOOLSET)) $@
  cmd_link = $(LINK.$(TOOLSET)) $(GYP_LDFLAGS) $(LDFLAGS.$(TOOLSET)) -o $@ $(LD_INPUTS) $(LIBS)
  
  quiet_cmd_solink = SOLINK($(TOOLSET)) $@
- cmd_solink = $(LINK.$(TOOLSET)) -shared $(GYP_LDFLAGS) $(LDFLAGS.$(TOOLSET)) -o $@ $(LD_INPUTS) $(LIBS)
+ cmd_solink = $(LINK.$(TOOLSET)) -shared $(GYP_LDFLAGS) $(LDFLAGS.$(TOOLSET)) -Wl,-soname=$(@F) -o $@ -Wl,--whole-archive $(LD_INPUTS) -Wl,--no-whole-archive $(LIBS)
  
  quiet_cmd_solink_module = SOLINK_MODULE($(TOOLSET)) $@
- cmd_solink_module = $(LINK.$(TOOLSET)) -shared $(GYP_LDFLAGS) $(LDFLAGS.$(TOOLSET)) -o $@ $(filter-out FORCE_DO_CMD, $^) $(LIBS)
+ cmd_solink_module = $(LINK.$(TOOLSET)) -shared $(GYP_LDFLAGS) $(LDFLAGS.$(TOOLSET)) -Wl,-soname=$(@F) -o $@ $(filter-out FORCE_DO_CMD, $^) $(LIBS)
  """
  
  
